### ISSUE TITLE
Augmented arithmetic assignments for immutable variables are not reported as errors

### ISSUE THREAD

**Issue by lancelote** [intellij-rust/intellij-rust#4567]
## Environment

* **IntelliJ Rust plugin version:** 0.2.109.2771-193-nightly
* **Rust toolchain version:** 1.38.0 (625451e37 2019-09-23) x86_64-apple-darwin
* **IDE name and version:** CLion 2020.1 Snapshot (CL-201.455)
* **Operating system:** macOS 10.15

## Problem description

Augmented arithmetic assignments are not checked for a variable mutability leading to a false negative.

## Steps to reproduce

```rust
fn bar() {
    let n = 1;

    n += 1;     // Fail: no error
    n = n + 1;  // Ok: error
}
```

<img width="338" alt="Screen Shot 2019-10-25 at 18 54 31" src="https://user-images.githubusercontent.com/2500320/67585758-e6a0ee80-f758-11e9-85e5-1bd210bd8dd4.png">
